src/server/bitops_family.cc

```diff
@@ -0,0 +1,302 @@
+    return (*cntx)->SendError(kInvalidIntErr);
+  }
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    return ReadValueBitsetAt(t->GetOpArgs(shard), key, offset);
+  };
+  Transaction* trans = cntx->transaction;
+  OpResult<bool> result = trans->ScheduleSingleHopT(std::move(cb));
+
+  if (result) {
+    DVLOG(2) << "GET" << trans->DebugId() << "': key: '" << key << ", value '" << result.value()
+             << "'\n";
+    // we have the value, now we need to get the bit at the location
+    long val = result.value() ? 1 : 0;
+    (*cntx)->SendLong(val);
+  } else {
+    switch (result.status()) {
+      case OpStatus::WRONG_TYPE:
+        (*cntx)->SendError(kWrongTypeErr);
+        break;
+      default:
+        DVLOG(2) << "GET " << key << " nil";
+        (*cntx)->SendLong(0);  // in case we don't have the value we should just send 0
+    }
+  }
+}
+
+void SetBit(CmdArgList args, ConnectionContext* cntx) {
+  // Support for the command "SETBIT key offset new_value"
+  // see https://redis.io/commands/setbit/
+
+  uint32_t offset{0};
+  int32_t value{0};
+  std::string_view key = ArgS(args, 1);
+
+  if (!absl::SimpleAtoi(ArgS(args, 2), &offset) || !absl::SimpleAtoi(ArgS(args, 3), &value)) {
+    return (*cntx)->SendError(kInvalidIntErr);
+  }
+
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    OverrideValue set_operation{t->GetOpArgs(shard)};
+
+    return set_operation.Set(key, offset, value != 0);
+  };
+
+  Transaction* trans = cntx->transaction;
+  OpResult<bool> result = trans->ScheduleSingleHopT(std::move(cb));
+  if (result) {
+    long res = result.value() ? 1 : 0;
+    (*cntx)->SendLong(res);
+  } else {
+    switch (result.status()) {
+      case OpStatus::WRONG_TYPE:
+        (*cntx)->SendError(kWrongTypeErr);
+        break;
+      case OpStatus::OUT_OF_MEMORY:
+        (*cntx)->SendError(kOutOfMemory);
+        break;
+      default:
+        DVLOG(2) << "SETBIT " << key << " nil" << result.status();
+        (*cntx)->SendLong(0);  // in case we don't have the value we should just send 0
+        break;
+    }
+  }
+}
+
+// ------------------------------------------------------------------------- //
+// This are the "callbacks" that we're using from above
+std::string GetString(EngineShard* shard, const PrimeValue& pv) {
+  std::string res;
+  if (pv.IsExternal()) {
+    auto* tiered = shard->tiered_storage();
+    auto [offset, size] = pv.GetExternalPtr();
+    res.resize(size);
+
+    std::error_code ec = tiered->Read(offset, size, res.data());
+    CHECK(!ec) << "TBD: " << ec;
+  } else {
+    pv.GetString(&res);
+  }
+
+  return res;
+}
+
+OpResult<bool> ReadValueBitsetAt(const OpArgs& op_args, std::string_view key, uint32_t offset) {
+  OpResult<std::string> result = ReadValue(op_args, key);
+  if (result) {
+    return GetBitValueSafe(result.value(), offset);
+  } else {
+    return result.status();
+  }
+}
+
+OpResult<std::string> ReadValue(const OpArgs& op_args, std::string_view key) {
```

Comment:
nit: you use `ReadValue` only once. 
It's matter of style of course but I do not see much value to divide 10 line function (ReadValueBitsetAt) into 2 5 line functions.

Reply:
since this code is going to have more functions added to it, if I don't see the value for something like it - not re-usable, I will merge them, You're right about this